trunk/SUAVE/Components/Energy/Networks/Battery_Cell_Cycler.py

```diff
@@ -0,0 +1,362 @@
+    
+            Assumptions:
+            Caps the throttle at 110% and linearly interpolates thrust off that
+    
+            Source:
+            N/A
+    
+            Inputs:
+            state [state()]
+    
+            Outputs: 
+            conditions.propulsion: 
+                current                       [amps]
+                battery_power_draw            [watts]
+                battery_energy                [joules]
+                battery_voltage_open_circuit  [volts]
+                battery_voltage_under_load    [volts]  
+    
+            Properties Used:
+            Defaulted values
+        """          
+    
+        # unpack
+        conditions        = state.conditions
+        numerics          = state.numerics 
+        avionics          = self.avionics 
+        battery           = self.battery   
+        D                 = numerics.time.differentiate    
+        I                 = numerics.time.integrate 
+        battery_data      = battery.discharge_performance_map 
+         
+        # Set battery energy
+        battery.current_energy      = conditions.propulsion.battery_energy
+        battery.pack_temperature    = conditions.propulsion.battery_pack_temperature
+        battery.charge_throughput   = conditions.propulsion.battery_charge_throughput     
+        battery.age_in_days         = conditions.propulsion.battery_age_in_days 
+        discharge_flag              = conditions.propulsion.battery_discharge    
+        battery.R_growth_factor     = conditions.propulsion.battery_resistance_growth_factor
+        battery.E_growth_factor     = conditions.propulsion.battery_capacity_fade_factor 
+        battery.max_energy          = conditions.propulsion.battery_max_aged_energy
+        V_th0                       = conditions.propulsion.battery_initial_thevenin_voltage
+        n_series                    = battery.pack_config.series  
+        n_parallel                  = battery.pack_config.parallel 
+        n_total                     = n_series*n_parallel
+    
+        # update ambient temperature based on altitude
+        battery.ambient_temperature                   = conditions.freestream.temperature   
+        battery.ambient_pressure                      = conditions.freestream.pressure
+        battery.cooling_fluid.thermal_conductivity    = conditions.freestream.thermal_conductivity
+        battery.cooling_fluid.kinematic_viscosity     = conditions.freestream.kinematic_viscosity
+        battery.cooling_fluid.density                 = conditions.freestream.density 
+         
+        #-------------------------------------------------------------------------------
+        # Predict Voltage and Battery Properties Depending on Battery Chemistry
+        #-------------------------------------------------------------------------------      
+        if type(battery) == SUAVE.Components.Energy.Storages.Batteries.Constant_Mass.Lithium_Ion_LiFePO4_18650:
```

Comment:
Can we update this so it calls the battery type to get the needed data?

Reply:
packaged into one line that is passed to the specific cell where the computations are done